tests/test_cli_utils.py

```diff
@@ -11,6 +12,235 @@
         ("samples/pipeline_template_examples/testing/example.yaml", False),
     ],
 )
-def test_is_sweep_pipeline(pipeline_file: Path, expected_is_sweep_pipeline: bool):
+def test_is_sweep_pipeline_file(pipeline_file: Path, expected_is_sweep_pipeline: bool):
+    """Test is_sweep_pipeline with file paths"""
     pipeline_file_path = Path(__file__).parent / pipeline_file
     assert is_sweep_pipeline(pipeline_file_path) is expected_is_sweep_pipeline
+
+
+def test_is_sweep_pipeline_dict_with_range():
+    """Test is_sweep_pipeline with a Python dict containing a sweep range"""
+    pipeline_dict = '''[
+        {
+            "method": "normalize",
+            "module_path": "httomolibgpu.prep.normalize",
+            "parameters": {
+                "cutoff": {
+                    "start": 10,
+                    "stop": 15,
+                    "step": 2
+                },
+                "minus_log": true
+            }
+        }
+    ]'''
+    assert is_sweep_pipeline(pipeline_dict) is True
+
+
+def test_is_sweep_pipeline_dict_with_list():
+    """Test is_sweep_pipeline with a Python dict containing a sweep list"""
+    pipeline_dict = '''[
+        {
+            "method": "normalize",
+            "module_path": "httomolibgpu.prep.normalize",
+            "parameters": {
+                "cutoff": [5, 10, 15],
+                "minus_log": true
+            }
+        }
+    ]'''
+    assert is_sweep_pipeline(pipeline_dict) is True
+
+
+def test_is_sweep_pipeline_json_string_with_sweep():
+    """Test is_sweep_pipeline with JSON strings containing both types of sweeps"""
+    # Test with sweep range
+    pipeline_range = '''[
+        {
+            "method": "normalize",
+            "module_path": "httomolibgpu.prep.normalize",
+            "parameters": {
+                "cutoff": {
+                    "start": 10,
+                    "stop": 15,
+                    "step": 2
+                }
+            }
+        }
+    ]'''
+    assert is_sweep_pipeline(pipeline_range) is True
+    
+    # Test with sweep list
+    pipeline_list = '''[
+        {
+            "method": "normalize",
+            "module_path": "httomolibgpu.prep.normalize",
+            "parameters": {
+                "cutoff": [5, 10, 15]
+            }
+        }
+    ]'''
+    assert is_sweep_pipeline(pipeline_list) is True
+
+
+def test_is_sweep_pipeline_real_world_example():
```

Comment:
I think this test can be removed, it's not really testing anything that isn't already being tested I don't think. Because it's not testing anything different from what current tests already check, I don't see any benefit from having this test (but please do say so if I'm missing something!).

Reply:
well no you are not missing anything, this test has to be removed as well as there is nothing new to test, the reason I made it was for my own double check to see if for a full working pipeline it will work as well not just bits of methods, got rid of it :)